NewsFeed/Common/Views/ScrollableSegmentedControl.swift

```diff
@@ -0,0 +1,235 @@
+//  ScrollableSegmentedControl
+//
+//  Created by MArko Satlan on 22/07/2018.
+//  Copyright © 2018 MArko Satlan. All rights reserved.
+//
+
+import UIKit
+
+class ScrollableSegmentedControl: UIControl {
+
+// MARK: - Properties
+    // Open
+    open var font = UIFont(name: "Georgia", size: 15) {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var buttonPadding: CGFloat = 15 {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var buttonMargin: CGFloat = 15 {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var selectedSegmentIndex: Int = -1 {
+        didSet {
+            highlightSelectedSegment()
+            calculateAndSetScrollOffset()
+            
+            // deselect previous button if different button is tapped
+            if oldValue != selectedSegmentIndex {
+                let previousButton = buttonsArray.filter{$0.tag == oldValue}.first
+                previousButton?.isSelected = false
+                UIButton.animate(withDuration: 0.3, animations: {
+                    previousButton?.transform = CGAffineTransform.identity
+                })
+            }
+        }
+    }
+    
+    // Private
+    private var buttonTitles: [String] = []
+    private var buttonsArray: [UIButton] = []
+    private let scrollView = UIScrollView()
+    
+// MARK: - Methods
+    // Init
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        
+        scrollView.showsHorizontalScrollIndicator = false
+        addSubview(scrollView)
+    }
+    
+    public init(items: [String]) {
+        self.buttonTitles = items
+        
+        super.init(frame: CGRect.zero)
+        
+        scrollView.showsHorizontalScrollIndicator = false
+        addSubview(scrollView)
+        createButtons()
+    }
+    
+    public func setItems(_ items: [String]) {
+        self.buttonTitles = items
+        createButtons()
+        setNeedsLayout()
+    }
+    
+    // Layout subviews - set frames
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        scrollView.frame = CGRect(x: 0, y: 0, width: frame.size.width, height: frame.size.height)
+        
+        updateUI()
+        drawSeparatorLine()
+        highlightSelectedSegment()
+        calculateAndSetScrollOffset()
+    }
+    
+    // Action methods
+    @objc private func buttonTapped(_ sender: UIButton) {
+        let previousButton = buttonsArray.filter{$0.tag == selectedSegmentIndex}.first
+        
+        selectedSegmentIndex = sender.tag
+        
+        if previousButton?.tag != sender.tag {
+            sendActions(for: .valueChanged)
+            previousButton?.isSelected = false
+            UIButton.animate(withDuration: 0.3, animations: {
+                previousButton?.transform = CGAffineTransform.identity
+            })
+        }
+ 
+        sender.isSelected = true
+    }
+    
+// Private
+    // Create buttons
```

Comment:
The idea was to split // MARK: - Methods into smaller sections by method type - i.e. // Init, // Layout subviews,  // Action methods, // Private methods - or maybe some other name would be more appropriate. Didn't put //MARK: - because it would be too many pragma marks

Reply:
I see, that's fine. Too many pragma marks in a small class don't improve code readability.

However, comments like `// Create buttons` and now `// Set up`, which don't explain what the method is doing but rather just repeating the method name, are redundant.